Services/AssessmentQuestion/examples/class.exTestPlayerGUI.php

```diff
@@ -0,0 +1,196 @@
+<?php
+
+/**
+ * When a component wants to integrate the assessment question service to present questions
+ * to users in an assessment scenario, the following use cases needs to be handled by the component.
+ * 
+ * This kind consume of the assessment question service does not support the offline export presentation yet.
+ */
+class exTestPlayerGUI
+{
+	/**
+	 * When presenting an assessment question to a user, the ilAsqQuestionPresentation provides the
+	 * interface methods to either get a renerable UI coponent for a question presentation or
+	 * for a solution presentation. An instance implementing the ilAsqPresentation according to
+	 * the given question type is provided by a factory method within ilAsqFactory.
+	 * 
+	 * Both methods getQuestionPresentation and getSolutionPresentation gets an instance of ilAsqSolution injected.
+	 * This solution can either be an user solution or the best solution.
+	 * 
+	 * Additional interface methods in ilAsqQuestionPresentation return renderable UI components for the generic
+	 * and specific feedbacks. These methods also need to get an ilAsqSolution instance injected.
+	 * 
+	 * In case of presenting the best solution the required ilAsqSolution instance can be retrieved
+	 * from the ilAsqQuestion interface.
+	 * 
+	 * For any existing user response that is to be presented with the question the instance
+	 * implementing ilAsqSolution can be requested from the ilAsqFactory using the solutionId
+	 * that is to be registered within the consuming component in relation to the user's id
+	 * and additional information (like e.g. test results).
+	 * 
+	 * Variants in usage defined by the consumer:
+	 * - question can be shown writable for the examine
+	 * - question can be shown readable for the examine
+	 * - feedbacks can be shown if required
+	 * - best solution can be shown if required
+	 */
+	public function showQuestion()
+	{		
+		$factory = new ilAsqFactory();
+		
+		$questionId = 0; // initialise with id of question to be shown
+		
+		/**
+		 * fetch possibly existing participant solution, an empty one is required otherwise
+		 */
+		
+		$participantSolution = $this->getParticipantSolution($questionId);
+		
+		/**
+		 * question presentation to be answered by the examine
+		 */
+		
+		$questionInstance = $factory->getQuestionInstance($questionId);
+		$questionPresentationGUI = $factory->getQuestionPresentationInstance($questionInstance);
```

Comment:
Maybe also move the purpose up here to get slimmer presentation classes (one per purpose).